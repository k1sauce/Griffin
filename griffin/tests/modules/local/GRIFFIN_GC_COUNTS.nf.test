nextflow_process {

    name "Test Process GRIFFIN_GC_COUNTS"
    script "modules/local/GRIFFIN_GC_COUNTS.nf"
    process "GRIFFIN_GC_COUNTS"
    tag "GRIFFIN_GC_COUNTS"

    //
    test("GRIFFIN_GC_COUNTS_T01") {
        when {
            process {
                /*
                tuple val(meta), path(bam), path(bai)
                val mappable_regions_path
                val ref_seq
                val chrom_sizes
                val map_q
                val size_range_low
                val size_range_high
                */
                """
                input[0] = [
                    [id: 'Healthy_demo', bam_file_name: 'Healthy_demo'], 
                    file("${projectDir}/test_data/bam/Healthy_GSM1833219_downsampled.sorted.mini.bam"),
                    file("${projectDir}/test_data/bam/Healthy_GSM1833219_downsampled.sorted.mini.bam.bai"),
                ]
                input[1] = file("${projectDir}/test_data/ref/k100_minus_exclusion_lists.mappable_regions.hg38.bed")
                input[2] = file("${projectDir}/test_data/ref/hg38.fa")
                input[3] = file("${projectDir}/test_data/ref/hg38.standard.chrom.sizes")
                input[4] = 20
                input[5] = 15
                input[6] = 500
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out.result).match("GRIFFIN_GC_COUNTS_T01")
        }
    }

}