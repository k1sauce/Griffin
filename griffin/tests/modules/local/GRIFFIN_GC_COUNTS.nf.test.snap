{
    "GRIFFIN_GC_COUNTS_T01": {
        "content": null,
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "25.04.2"
        },
        "timestamp": "2025-05-28T21:14:24.337534"
    }
}